src/Backend/Modules/Blog/Engine/Model.php

```diff
@@ -897,38 +892,31 @@ public static function insertCategory(array $item, array $meta = null): int
         return $item['id'];
     }
 
-    /**
-     * Inserts a new comment (Taken from FrontendBlogModel)
-     *
-     * @param array $comment The comment to add.
-     *
-     * @return int
-     */
-    public static function insertComment(array $comment): int
+    public static function insertComment(array $data): int
     {
-        // get database
-        $database = BackendModel::getContainer()->get('database');
+        $entityManager = BackendModel::get('doctrine.orm.default_entity_manager');
+
+        $comment = new Comment(
+            $data['post_id'],
+            Locale::fromString($data['language']),
+            $data['author'],
+            $data['email'],
+            $data['website'],
+            $data['text'],
+            'comment',
+            $data['status'],
+            $data['data']
+        );
 
-        // insert comment
-        $comment['id'] = (int) $database->insert('blog_comments', $comment);
+        $entityManager->persist($comment);
+        $entityManager->flush($comment);
 
         // recalculate if published
-        if ($comment['status'] == 'published') {
-            // num comments
-            $numComments = (int) BackendModel::getContainer()->get('database')->getVar(
-                'SELECT COUNT(i.id) AS comment_count
-                 FROM blog_comments AS i
-                 INNER JOIN blog_posts AS p ON i.post_id = p.id AND i.language = p.language
-                 WHERE i.status = ? AND i.post_id = ? AND i.language = ? AND p.status = ?
-                 GROUP BY i.post_id',
-                ['published', $comment['post_id'], BL::getWorkingLanguage(), 'active']
-            );
-
-            // update num comments
-            $database->update('blog_posts', ['num_comments' => $numComments], 'id = ?', $comment['post_id']);
+        if ($comment->getStatus() == 'published') {
```

Comment:
===

Reply:
Fixed in 18adc30c0965878417a31b49881be34846e4911a